components/ws-manager/pkg/manager/manager.go

```diff
@@ -513,50 +520,6 @@ func (m *Manager) restoreVolumeSnapshotFromHandle(ctx context.Context, id, handl
 	return nil
 }
 
-func (m *Manager) DeleteVolumeSnapshot(ctx context.Context, req *api.DeleteVolumeSnapshotRequest) (res *api.DeleteVolumeSnapshotResponse, err error) {
```

Comment:
Why was this deleted? It is needed for GC to work.

Reply:
@jenting I reverted deletion of this function, I assume it was done by mistake. Reverted since you will be OOO until next week, and I would like to get this merged prior to that. LMK.

Reply:
Yes, it was a mistake.

Reply:
We need to pass the workspace type to the function `restore`.

Reply:
ah. hmm. that will require a bit more changes then, as I need to make webapp change to propagate it all the way from there (and hopefully that will not be too complicated). And it seems like that will require DB change as well to save what workspace type that volume snapshot belongs to.
So I will do that tomorrow morning.

Reply:
If it's not a soft delete (i.e., it's a hard delete), I think we don't have to restore the VolumeSnapshot form handler and delete the VolumeSnapshot after that.

Therefore, we don't have to change the WebApp code and DB schema.

WDYT? @sagor999

Reply:
we have to, as otherwise we will not delete snapshot/image from gcp.